src/warc_types.rs

```diff
@@ -2,55 +2,81 @@ use crate::parser;
 use crate::{Error, Record};
 use std::fs;
 use std::io;
-use std::io::{BufRead, BufReader, Seek, SeekFrom, Write};
+use std::io::{BufRead, BufReader, BufWriter, Write};
 use std::path::Path;
 
 const KB: usize = 1_024;
 const MB: usize = 1_048_576;
 
-pub struct File {
-    reader: BufReader<fs::File>,
+pub struct WarcReader<R> {
+    reader: R,
 }
 
-impl File {
-    pub fn open<P: AsRef<Path>>(path: P) -> io::Result<Self> {
-        let file = fs::OpenOptions::new()
-            .read(true)
-            .write(true)
-            .create(true)
-            .open(&path)?;
-        let reader = BufReader::with_capacity(1 * MB, file);
+pub struct WarcWriter<W> {
+    writer: W,
+}
 
-        Ok(File { reader: reader })
+impl<W: Write> WarcWriter<W> {
+    pub fn new(w: W) -> Self {
+        WarcWriter { writer: w }
     }
 
     pub fn write(&mut self, record: &Record) -> io::Result<usize> {
         let mut bytes_written = 0;
 
-        let mut file = self.reader.get_ref();
-        file.seek(SeekFrom::End(0))?;
```

Comment:
I just realized that removing this breaks some expected behavior. Without this, the writer writes over the beginning of the file. https://github.com/jedireza/warc/issues/14